hedera-node/hapi-utils/src/main/java/com/hedera/node/app/hapi/utils/CommonUtils.java

```diff
@@ -88,4 +137,65 @@ public static boolean productWouldOverflow(final long multiplier, final long mul
     static void setSha384HashTag(final String sha384HashTag) {
         CommonUtils.sha384HashTag = sha384HashTag;
     }
+
+    /**
+     * check TransactionBody and return the HederaFunctionality
+     *
+     * @param txn the {@code TransactionBody}
+     * @return one of HederaFunctionality
+     * @throws UnknownHederaFunctionality if all the check fails
+     */
+    @NonNull
+    public static HederaFunctionality functionOf(@NonNull final TransactionBody txn) throws UnknownHederaFunctionality {
+        DataCase dataCase = txn.getDataCase();
+
+        return switch (dataCase) {
+            case CONTRACTCALL -> ContractCall;
+            case CONTRACTCREATEINSTANCE -> ContractCreate;
+            case CONTRACTUPDATEINSTANCE -> ContractUpdate;
+            case CONTRACTDELETEINSTANCE -> ContractDelete;
+            case ETHEREUMTRANSACTION -> EthereumTransaction;
+            case CRYPTOADDLIVEHASH -> CryptoAddLiveHash;
+            case CRYPTOAPPROVEALLOWANCE -> CryptoApproveAllowance;
+            case CRYPTODELETEALLOWANCE -> CryptoDeleteAllowance;
+            case CRYPTOCREATEACCOUNT -> CryptoCreate;
+            case CRYPTODELETE -> CryptoDelete;
+            case CRYPTODELETELIVEHASH -> CryptoDeleteLiveHash;
+            case CRYPTOTRANSFER -> CryptoTransfer;
+            case CRYPTOUPDATEACCOUNT -> CryptoUpdate;
+            case FILEAPPEND -> FileAppend;
+            case FILECREATE -> FileCreate;
+            case FILEDELETE -> FileDelete;
+            case FILEUPDATE -> FileUpdate;
+            case SYSTEMDELETE -> SystemDelete;
+            case SYSTEMUNDELETE -> SystemUndelete;
+            case FREEZE -> Freeze;
+            case CONSENSUSCREATETOPIC -> ConsensusCreateTopic;
+            case CONSENSUSUPDATETOPIC -> ConsensusUpdateTopic;
+            case CONSENSUSDELETETOPIC -> ConsensusDeleteTopic;
+            case CONSENSUSSUBMITMESSAGE -> ConsensusSubmitMessage;
+            case UNCHECKEDSUBMIT -> UncheckedSubmit;
+            case TOKENCREATION -> TokenCreate;
+            case TOKENFREEZE -> TokenFreezeAccount;
+            case TOKENUNFREEZE -> TokenUnfreezeAccount;
+            case TOKENGRANTKYC -> TokenGrantKycToAccount;
+            case TOKENREVOKEKYC -> TokenRevokeKycFromAccount;
+            case TOKENDELETION -> TokenDelete;
+            case TOKENUPDATE -> TokenUpdate;
+            case TOKENMINT -> TokenMint;
+            case TOKENBURN -> TokenBurn;
+            case TOKENWIPE -> TokenAccountWipe;
+            case TOKENASSOCIATE -> TokenAssociateToAccount;
+            case TOKENDISSOCIATE -> TokenDissociateFromAccount;
+            case TOKEN_FEE_SCHEDULE_UPDATE -> TokenFeeScheduleUpdate;
+            case TOKEN_PAUSE -> TokenPause;
+            case TOKEN_UNPAUSE -> TokenUnpause;
+            case SCHEDULECREATE -> ScheduleCreate;
+            case SCHEDULEDELETE -> ScheduleDelete;
+            case SCHEDULESIGN -> ScheduleSign;
+            case NODE_STAKE_UPDATE -> NodeStakeUpdate;
```

Comment:
⚠️ This changes behavior compared to `MiscUtils.functionOf()`, which throws on a `NODE_STAKE_UPDATE`.

We should port unit tests from `MiscUtilsTest` and make sure no behavior changes, since this is changing production code.

Reply:
removed NODE_STAKE_UPDATE from here.